### ISSUE TITLE
Error raised for create_openmm_system() (AM1-BCC charges) from molecules with net charge != 0

### ISSUE THREAD

**Issue by Kael-the-Mad** [openforcefield/openforcefield#627]
**Describe the bug**
I have adapted the "1_cyclohexane_1_ethanol" start-up python code and use it on other molecules.
One molecule is rosiglitazone (drug-like), whose net charge = 0, and the adapted script worked. Then, I tried other molecules, and some failed. A simple example is 'CCCCCCCCC(=O)[O-]' (or simpler ones).
 

**To Reproduce**
The adapted code is:
```
from simtk.openmm.app import PDBFile
from openforcefield.topology import Molecule
from openforcefield.topology import Topology
from openforcefield.typing.engines.smirnoff import ForceField
from simtk.openmm import XmlSerializer
from glob import glob
import time
from simtk import openmm, unit

sdfs = glob("m*.sdf")
for sdf in sdfs:
	nym = sdf[:-4]
	print(nym)
	# PDBFile provide omm topoplogy and initial positions
	# [kael: do we really need a pdb here?]
	pdbfile = PDBFile('j'+nym+'.pdb')
	# here we read a sdf for the correct top and aromaticity
	xBinder = Molecule.from_file(sdf)
	# Create the Open Force Field Topology from an OpenMM Topology object.
	# [kael: is this the most magic step? No! But, still magic]
	omm_topology = pdbfile.topology
	off_topology = Topology.from_openmm(omm_topology, unique_molecules=[xBinder])
	# Load the OpenFF "Parsley" force field.
	forcefield = ForceField('openff-1.0.0.offxml')
	# SMIRKS matching
	if True:
		with open('inspect_'+nym+'.txt','w',encoding='utf-8') as g:
			molecule_force_list = forcefield.label_molecules(off_topology)
			# Print out a formatted description of the parameters applied to this molecule
			for mol_idx, mol_forces in enumerate(molecule_force_list):
				print(f'Forces for molecule {mol_idx}', file=g)
				for force_tag, force_dict in mol_forces.items():
					print(f"\n{force_tag}:", file=g)
					for (atom_indices, parameter) in force_dict.items():
						atomstr=''
						for idx in atom_indices:
							atomstr += '%3s' % idx
						print("atoms: %s  parameter_id: %s  smirks %s" % (atomstr, parameter.id, parameter.smirks) , file=g)
	#
	# Parametrize the topology and create an OpenMM System.
	system = forcefield.create_openmm_system(off_topology, charge_from_molecules) # on this step the amber ACC is triggered
	with open('system_'+nym+'.xml', 'w') as f:
		f.write(XmlSerializer.serialize(system))
	#
	"""
	with open('system_BRL.xml', 'r') as f:
		system2 = XmlSerializer.deserialize(f.read())
	"""
	#Run a simulation#
	#We can now use the System object to run molecular dynamics trajectories with OpenMM.
	# Propagate the System with Langevin dynamics.
	time_step = 2*unit.femtoseconds  # simulation timestep
	temperature = 300*unit.kelvin  # simulation temperature
	friction = 1/unit.picosecond  # collision rate
	integrator = openmm.LangevinIntegrator(temperature, friction, time_step)
	# Length of the simulation.
	num_steps = 10000  # number of integration steps to run
	# Logging options.
	trj_freq = 100  # number of steps per written trajectory frame
	data_freq = 100  # number of steps per written simulation statistics
	# Set up an OpenMM simulation.
	simulation = openmm.app.Simulation(omm_topology, system, integrator)
	# Set the initial positions.
	positions = pdbfile.getPositions() 
	simulation.context.setPositions(positions)
	# Randomize the velocities from a Boltzmann distribution at a given temperature.
	simulation.context.setVelocitiesToTemperature(temperature)
	# Configure the information in the output files.
	pdb_reporter = openmm.app.PDBReporter('traj_'+nym+'.pdb', trj_freq)
	state_data_reporter = openmm.app.StateDataReporter('data_'+nym+'.csv', data_freq, step=True,
													potentialEnergy=True, temperature=True,
													density=True)
	simulation.reporters.append(pdb_reporter)
	simulation.reporters.append(state_data_reporter)
	print("Starting simulation")
	start = time.process_time()
	# Run the simulation
	simulation.step(num_steps)
	end = time.process_time()
	print("Elapsed time %.2f seconds" % (end-start))
	print("Done!")


#
```

If the problem involves a specific molecule or file, please upload that as well.
Required files to reproduce the error include:
1. "jmA~KNA_1@3sz1A.pdb" whose content is
```
ATOM      1  C1  UNL    1      15.965 -26.039   8.086  1.00  0.00           C  
ATOM      2  O1  UNL    1      16.193 -25.641   6.946  1.00  0.00           O  
ATOM      3  C2  UNL    1      15.510 -25.062   9.165  1.00  0.00           C  
ATOM      4  O2  UNL    1      16.116 -27.367   8.369  1.00  0.00           O  
ATOM      5  C3  UNL    1      15.468 -25.726  10.556  1.00  0.00           C  
ATOM      6  C4  UNL    1      14.217 -25.299  11.315  1.00  0.00           C  
ATOM      7  C5  UNL    1      14.187 -25.946  12.704  1.00  0.00           C  
ATOM      8  C6  UNL    1      14.095 -27.489  12.689  1.00  0.00           C  
ATOM      9  C7  UNL    1      13.997 -28.037  14.145  1.00  0.00           C  
ATOM     10  C8  UNL    1      15.117 -29.064  14.401  1.00  0.00           C  
ATOM     11  C9  UNL    1      14.855 -29.910  15.667  1.00  0.00           C  
ATOM     12  H1  UNL    1      14.531 -24.707   8.921  1.00  0.00           H  
ATOM     13  H2  UNL    1      16.212 -24.256   9.202  1.00  0.00           H  
ATOM     14  H3  UNL    1      16.334 -25.433  11.112  1.00  0.00           H  
ATOM     15  H4  UNL    1      15.451 -26.789  10.431  1.00  0.00           H  
ATOM     16  H5  UNL    1      13.351 -25.603  10.766  1.00  0.00           H  
ATOM     17  H6  UNL    1      14.226 -24.235  11.428  1.00  0.00           H  
ATOM     18  H7  UNL    1      13.338 -25.565  13.232  1.00  0.00           H  
ATOM     19  H8  UNL    1      15.116 -25.701  13.175  1.00  0.00           H  
ATOM     20  H9  UNL    1      14.968 -27.890  12.218  1.00  0.00           H  
ATOM     21  H10 UNL    1      13.220 -27.781  12.146  1.00  0.00           H  
ATOM     22  H11 UNL    1      13.046 -28.509  14.282  1.00  0.00           H  
ATOM     23  H12 UNL    1      14.103 -27.224  14.833  1.00  0.00           H  
ATOM     24  H13 UNL    1      16.043 -28.541  14.521  1.00  0.00           H  
ATOM     25  H14 UNL    1      15.151 -29.729  13.563  1.00  0.00           H  
ATOM     26  H15 UNL    1      14.816 -29.269  16.523  1.00  0.00           H  
ATOM     27  H16 UNL    1      15.645 -30.622  15.790  1.00  0.00           H  
ATOM     28  H17 UNL    1      13.923 -30.426  15.565  1.00  0.00           H  
CONECT    1    4    3    2    2
CONECT    2    1    1
CONECT    3    1    5   12   13
CONECT    4    1
CONECT    5    3    6   14   15
CONECT    6    5    7   16   17
CONECT    7    6   18   19    8
CONECT    8    7    9   20   21
CONECT    9    8   10   22   23
CONECT   10    9   11   24   25
CONECT   11   10   26   27   28
CONECT   12    3
CONECT   13    3
CONECT   14    5
CONECT   15    5
CONECT   16    6
CONECT   17    6
CONECT   18    7
CONECT   19    7
CONECT   20    8
CONECT   21    8
CONECT   22    9
CONECT   23    9
CONECT   24   10
CONECT   25   10
CONECT   26   11
CONECT   27   11
CONECT   28   11

```
2. "mA~KNA_1@3sz1A.sdf"
```
A~KNA_1@3sz1A.pdb
     RDKit          3D

 28 27  0  0  0  0  0  0  0  0999 V2000
   15.9650  -26.0390    8.0860 C   0  0  0  0  0  0  0  0  0  0  0  0
   16.1930  -25.6410    6.9460 O   0  0  0  0  0  0  0  0  0  0  0  0
   15.5100  -25.0620    9.1650 C   0  0  0  0  0  0  0  0  0  0  0  0
   16.1160  -27.3670    8.3690 O   0  0  0  0  0  0  0  0  0  0  0  0
   15.4680  -25.7260   10.5560 C   0  0  0  0  0  0  0  0  0  0  0  0
   14.2170  -25.2990   11.3150 C   0  0  0  0  0  0  0  0  0  0  0  0
   14.1870  -25.9460   12.7040 C   0  0  0  0  0  0  0  0  0  0  0  0
   14.0950  -27.4890   12.6890 C   0  0  0  0  0  0  0  0  0  0  0  0
   13.9970  -28.0370   14.1450 C   0  0  0  0  0  0  0  0  0  0  0  0
   15.1170  -29.0640   14.4010 C   0  0  0  0  0  0  0  0  0  0  0  0
   14.8550  -29.9100   15.6670 C   0  0  0  0  0  0  0  0  0  0  0  0
   14.5306  -24.7072    8.9206 H   0  0  0  0  0  0  0  0  0  0  0  0
   16.2123  -24.2556    9.2019 H   0  0  0  0  0  0  0  0  0  0  0  0
   16.3341  -25.4333   11.1120 H   0  0  0  0  0  0  0  0  0  0  0  0
   15.4508  -26.7886   10.4314 H   0  0  0  0  0  0  0  0  0  0  0  0
   13.3505  -25.6031   10.7658 H   0  0  0  0  0  0  0  0  0  0  0  0
   14.2263  -24.2350   11.4282 H   0  0  0  0  0  0  0  0  0  0  0  0
   13.3377  -25.5654   13.2320 H   0  0  0  0  0  0  0  0  0  0  0  0
   15.1159  -25.7005   13.1749 H   0  0  0  0  0  0  0  0  0  0  0  0
   14.9680  -27.8905   12.2183 H   0  0  0  0  0  0  0  0  0  0  0  0
   13.2202  -27.7812   12.1465 H   0  0  0  0  0  0  0  0  0  0  0  0
   13.0464  -28.5085   14.2824 H   0  0  0  0  0  0  0  0  0  0  0  0
   14.1032  -27.2243   14.8329 H   0  0  0  0  0  0  0  0  0  0  0  0
   16.0429  -28.5413   14.5214 H   0  0  0  0  0  0  0  0  0  0  0  0
   15.1510  -29.7286   13.5631 H   0  0  0  0  0  0  0  0  0  0  0  0
   14.8160  -29.2688   16.5227 H   0  0  0  0  0  0  0  0  0  0  0  0
   15.6446  -30.6215   15.7902 H   0  0  0  0  0  0  0  0  0  0  0  0
   13.9230  -30.4256   15.5649 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  4  1  0
  1  3  1  0
  2  1  2  0
  3  5  1  0
  3 12  1  0
  3 13  1  0
  5  6  1  0
  5 14  1  0
  5 15  1  0
  6  7  1  0
  6 16  1  0
  6 17  1  0
  7 18  1  0
  7 19  1  0
  8  7  1  0
  8  9  1  0
  8 20  1  0
  8 21  1  0
  9 10  1  0
  9 22  1  0
  9 23  1  0
 10 11  1  0
 10 24  1  0
 10 25  1  0
 11 26  1  0
 11 27  1  0
 11 28  1  0
M  CHG  1   4  -1
M  END

```

[KNA_pdb_sdf.zip](https://github.com/openforcefield/openforcefield/files/4834468/KNA_pdb_sdf.zip)

**Output**
```
Warning: Unable to load toolkit 'OpenEye Toolkit'. The Open Force Field Toolkit does not require the OpenEye Toolkits, and can use RDKit/AmberTools instead. However, if you have a valid license for the OpenEye Toolkits, consider installing them for faster performance and additional file format support: https://docs.eyesopen.com/toolkits/python/quickstart-python/linuxosx.html OpenEye offers free Toolkit licenses for academics: https://www.eyesopen.com/academic-licensing
Warning: In AmberToolsToolkitwrapper.compute_partial_charges_am1bcc: Molecule 'A~KNA_1@3sz1A.pdb' has more than one conformer, but this function will only generate charges for the first one.

Welcome to antechamber 19.0: molecular input file processor.

acdoctor mode is on: check and diagnose problems in the input file.
-- Check Format for sdf File --
   Status: pass
-- Check Unusual Elements --
   Status: pass
-- Check Open Valences --
   Status: pass
-- Check Geometry --
      for those bonded
      for those not bonded
   Status: pass
-- Check Weird Bonds --
   Status: pass
-- Check Number of Units --
   Status: pass
acdoctor mode has completed checking the input file.

Info: Total number of electrons: 88; net charge: -1

Running: /home/kael/openforcefield/bin/sqm -O -i sqm.in -o sqm.out

Welcome to antechamber 19.0: molecular input file processor.

acdoctor mode is on: check and diagnose problems in the input file.
-- Check Format for mol2 File --
   Status: pass
-- Check Unusual Elements --
   Status: pass
-- Check Open Valences --
   Status: pass
-- Check Geometry --
      for those bonded
      for those not bonded
   Status: pass
-- Check Weird Bonds --
antechamber: Fatal Error!
Weird atomic valence (3) for atom (ID: 1, Name: C1).
       Possible open valence.
mA~KNA_1@3sz1A
Traceback (most recent call last):
  File "2b.simulLigs.py", line 45, in <module>
    system = forcefield.create_openmm_system(off_topology) # on this step the amber ACC is triggered
  File "/home/kael/openforcefield/lib/python3.7/site-packages/openforcefield/typing/engines/smirnoff/forcefield.py", line 1138, in create_openmm_system
    parameter_handler.create_force(system, topology, **kwargs)
  File "/home/kael/openforcefield/lib/python3.7/site-packages/openforcefield/typing/engines/smirnoff/parameters.py", line 2939, in create_force
    temp_mol.compute_partial_charges_am1bcc(toolkit_registry=toolkit_registry)
  File "/home/kael/openforcefield/lib/python3.7/site-packages/openforcefield/topology/molecule.py", line 2014, in compute_partial_charges_am1bcc
    self
  File "/home/kael/openforcefield/lib/python3.7/site-packages/openforcefield/utils/toolkits.py", line 3025, in call
    return method(*args, **kwargs)
  File "/home/kael/openforcefield/lib/python3.7/site-packages/openforcefield/utils/toolkits.py", line 2762, in compute_partial_charges_am1bcc
    molecule.name, molecule.to_smiles()))
Exception: Antechamber/sqm partial charge calculation failed on molecule A~KNA_1@3sz1A.pdb (SMILES [H][C]([H])([H])[C]([H])([H])[C]([H])([H])[C]([H])([H])[C]([H])([H])[C]([H])([H])[C]([H])([H])[C]([H])([H])[C](=[O])[O-])
```

**Computing environment (please complete the following information):**
 - CentOS Linux release 7.7.1908 (Core)
 - Output of running `conda list` 
```
# packages in environment at /home/kael/openforcefield:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                      1_llvm    conda-forge
amberlite                 16.0                     pypi_0    pypi
ambertools                17.0                     pypi_0    pypi
attrs                     19.3.0                   pypi_0    pypi
backcall                  0.1.0                    pypi_0    pypi
bleach                    3.1.4                    pypi_0    pypi
boost                     1.72.0           py37h9de70de_0    conda-forge
boost-cpp                 1.72.0               h8e57a91_0    conda-forge
bson                      0.5.9                    pypi_0    pypi
bzip2                     1.0.8                h516909a_2    conda-forge
ca-certificates           2020.4.5.1           hecc5488_0    conda-forge
cairo                     1.16.0            hcf35c78_1003    conda-forge
certifi                   2020.4.5.1       py37hc8dfbb8_0    conda-forge
cffi                      1.14.0           py37hd463f26_0    conda-forge
chardet                   3.0.4           py37hc8dfbb8_1006    conda-forge
conda                     4.8.3            py37hc8dfbb8_1    conda-forge
conda-package-handling    1.6.0            py37h8f50634_2    conda-forge
cryptography              2.8              py37hb09aad4_2    conda-forge
curl                      7.69.1               h33f0ec9_0    conda-forge
cycler                    0.10.0                   pypi_0    pypi
cython                    0.29.16          py37h3340039_0    conda-forge
decorator                 4.4.2                    pypi_0    pypi
defusedxml                0.6.0                    pypi_0    pypi
entrypoints               0.3             py37hc8dfbb8_1001    conda-forge
fftw                      3.3.8           nompi_h7f3a6c3_1110    conda-forge
fftw3f                    3.3.4                         2    omnia
fontconfig                2.13.1            h86ecdb6_1001    conda-forge
freetype                  2.10.1               he06d7ca_0    conda-forge
gettext                   0.19.8.1          hc5be6a0_1002    conda-forge
glib                      2.58.3          py37he00f558_1004    conda-forge
hdf4                      4.2.13            hf30be14_1003    conda-forge
hdf5                      1.10.5          nompi_h3c11f04_1104    conda-forge
icu                       64.2                 he1b5a44_1    conda-forge
idna                      2.9                      pypi_0    pypi
importlib-metadata        1.6.0            py37hc8dfbb8_0    conda-forge
importlib_metadata        1.6.0                         0    conda-forge
ipykernel                 5.2.0            py37h43977f1_1    conda-forge
ipython                   7.13.0           py37hc8dfbb8_2    conda-forge
ipython-genutils          0.2.0                    pypi_0    pypi
ipython_genutils          0.2.0                      py_1    conda-forge
jedi                      0.17.0           py37hc8dfbb8_0    conda-forge
jinja2                    2.11.2                   pypi_0    pypi
jpeg                      9c                h14c3975_1001    conda-forge
jsonschema                3.2.0            py37hc8dfbb8_1    conda-forge
jupyter-client            6.1.3                    pypi_0    pypi
jupyter_client            6.1.3                      py_0    conda-forge
jupyter_core              4.6.3            py37hc8dfbb8_1    conda-forge
kiwisolver                1.2.0            py37h99015e2_0    conda-forge
krb5                      1.17.1               h2fd8d38_0    conda-forge
ld_impl_linux-64          2.34                 h53a641e_0    conda-forge
libblas                   3.8.0               16_openblas    conda-forge
libcblas                  3.8.0               16_openblas    conda-forge
libcurl                   7.69.1               hf7181ac_0    conda-forge
libedit                   3.1.20170329      hf8c457e_1001    conda-forge
libffi                    3.2.1             he1b5a44_1007    conda-forge
libgcc-ng                 9.2.0                h24d8f2e_2    conda-forge
libgfortran-ng            7.3.0                hdf63c60_5    conda-forge
libgomp                   9.2.0                h24d8f2e_2    conda-forge
libiconv                  1.15              h516909a_1006    conda-forge
liblapack                 3.8.0               16_openblas    conda-forge
libnetcdf                 4.7.4           nompi_h9f9fd6a_101    conda-forge
libopenblas               0.3.9                h5ec1e0e_0    conda-forge
libpng                    1.6.37               hed695b0_1    conda-forge
libsodium                 1.0.17               h516909a_0    conda-forge
libssh2                   1.8.2                h22169c7_2    conda-forge
libstdcxx-ng              9.2.0                hdf63c60_2    conda-forge
libtiff                   4.1.0                hc7e4089_6    conda-forge
libuuid                   2.32.1            h14c3975_1000    conda-forge
libwebp-base              1.1.0                h516909a_3    conda-forge
libxcb                    1.13              h14c3975_1002    conda-forge
libxml2                   2.9.10               hee79883_0    conda-forge
llvm-openmp               10.0.0               hc9558a2_0    conda-forge
lz4-c                     1.9.2                he1b5a44_0    conda-forge
markupsafe                1.1.1            py37h8f50634_1    conda-forge
matplotlib-base           3.2.1            py37h30547a4_0    conda-forge
mistune                   0.8.4           py37h8f50634_1001    conda-forge
mmpbsa-py                 16.0                     pypi_0    pypi
more-itertools            8.2.0                    pypi_0    pypi
msgpack-python            1.0.0            py37h99015e2_1    conda-forge
nbconvert                 5.6.1            py37hc8dfbb8_1    conda-forge
nbformat                  5.0.4                    pypi_0    pypi
ncurses                   6.1               hf484d3e_1002    conda-forge
netcdf-fortran            4.5.2           nompi_h45d7149_104    conda-forge
networkx                  2.4                      pypi_0    pypi
notebook                  6.0.3                    py37_0    conda-forge
numpy                     1.18.1           py37h8960a57_1    conda-forge
olefile                   0.46                     pypi_0    pypi
openforcefield            0.6.0                    py37_1    omnia
openforcefields           1.1.1                    py37_0    omnia
openmm                    7.4.1           py37_cuda101_rc_1    omnia
openssl                   1.1.1f               h516909a_0    conda-forge
packaging                 20.1                     pypi_0    pypi
pandas                    1.0.3            py37h0da4684_0    conda-forge
pandoc                    2.9.2.1                       0    conda-forge
pandocfilters             1.4.2                    pypi_0    pypi
parmed                    3.2.0                    pypi_0    pypi
parso                     0.7.0                    pypi_0    pypi
pcre                      8.44                 he1b5a44_0    conda-forge
pdb4amber                 1.7.dev0                 pypi_0    pypi
perl                      5.26.2            h516909a_1006    conda-forge
pexpect                   4.8.0            py37hc8dfbb8_1    conda-forge
pickleshare               0.7.5           py37hc8dfbb8_1001    conda-forge
pillow                    7.1.1            py37h718be6c_0    conda-forge
pip                       20.0.2                   pypi_0    pypi
pixman                    0.38.0            h516909a_1003    conda-forge
pluggy                    0.13.0                   py37_0    conda-forge
prometheus-client         0.7.1                    pypi_0    pypi
prometheus_client         0.7.1                      py_0    conda-forge
prompt-toolkit            3.0.5                    pypi_0    pypi
pthread-stubs             0.4               h14c3975_1001    conda-forge
ptyprocess                0.6.0                    pypi_0    pypi
py                        1.8.1                    pypi_0    pypi
pycairo                   1.19.1           py37h01af8b0_3    conda-forge
pycosat                   0.6.3           py37h8f50634_1004    conda-forge
pycparser                 2.20                     pypi_0    pypi
pygments                  2.6.1                    pypi_0    pypi
pyopenssl                 19.1.0                   pypi_0    pypi
pyparsing                 2.4.7                    pypi_0    pypi
pyrsistent                0.16.0           py37h8f50634_0    conda-forge
pysocks                   1.7.1            py37hc8dfbb8_1    conda-forge
pytest                    5.4.1            py37hc8dfbb8_0    conda-forge
python                    3.7.6           h8356626_5_cpython    conda-forge
python-dateutil           2.8.1                    pypi_0    pypi
python_abi                3.7                     1_cp37m    conda-forge
pytraj                    2.0.5                    pypi_0    pypi
pytz                      2019.3                   pypi_0    pypi
pyyaml                    5.3.1            py37h8f50634_0    conda-forge
pyzmq                     19.0.0           py37hac76be4_1    conda-forge
rdkit                     2020.03.1        py37hdd87690_1    conda-forge
readline                  8.0                  hf8c457e_0    conda-forge
requests                  2.23.0                   pypi_0    pypi
ruamel_yaml               0.15.80         py37h8f50634_1001    conda-forge
sander                    16.0                     pypi_0    pypi
scipy                     1.4.1            py37ha3d9a3c_3    conda-forge
send2trash                1.5.0                    pypi_0    pypi
setuptools                46.1.3           py37hc8dfbb8_0    conda-forge
six                       1.14.0                   pypi_0    pypi
smirnoff99frosst          1.1.0                    py37_1    omnia
sqlite                    3.30.1               hcee41ef_0    conda-forge
terminado                 0.8.3            py37hc8dfbb8_1    conda-forge
testpath                  0.4.4                    pypi_0    pypi
tk                        8.6.10               hed695b0_0    conda-forge
toml                      0.10.0                   pypi_0    pypi
tornado                   6.0.4            py37h8f50634_1    conda-forge
tqdm                      4.45.0                   pypi_0    pypi
traitlets                 4.3.3            py37hc8dfbb8_1    conda-forge
urllib3                   1.25.8           py37hc8dfbb8_1    conda-forge
wcwidth                   0.1.9                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
wheel                     0.34.2                   pypi_0    pypi
xmltodict                 0.12.0                   pypi_0    pypi
xorg-kbproto              1.0.7             h14c3975_1002    conda-forge
xorg-libice               1.0.10               h516909a_0    conda-forge
xorg-libsm                1.2.3             h84519dc_1000    conda-forge
xorg-libx11               1.6.9                h516909a_0    conda-forge
xorg-libxau               1.0.9                h14c3975_0    conda-forge
xorg-libxdmcp             1.1.3                h516909a_0    conda-forge
xorg-libxext              1.3.4                h516909a_0    conda-forge
xorg-libxrender           0.9.10            h516909a_1002    conda-forge
xorg-libxt                1.2.0                h516909a_0    conda-forge
xorg-renderproto          0.11.1            h14c3975_1002    conda-forge
xorg-xextproto            7.3.0             h14c3975_1002    conda-forge
xorg-xproto               7.0.31            h14c3975_1007    conda-forge
xz                        5.2.5                h516909a_0    conda-forge
yaml                      0.2.3                h516909a_0    conda-forge
zeromq                    4.3.2                he1b5a44_2    conda-forge
zipp                      3.1.0                    pypi_0    pypi
zlib                      1.2.11            h516909a_1006    conda-forge
zstd                      1.4.4                h6597ccf_3    conda-forge
```

**Additional context**
According to the openforcefield.utils.toolkits.py, `class AmberToolsToolkitWrapper`, `compute_partial_charges_am1bcc()` method,
I have tried to directly use antechamber to calculate the AM1-BCC charges.
firstly, 
```
                os.system(
                    "antechamber -i molecule.sdf -fi sdf -o charged.mol2 -fo mol2 -pf "
                    "yes -c bcc -nc {}".format(net_charge))
```
 This was successfully done.
```
however, the next
                # Write out just charges
                os.system(
                    "antechamber -i charged.mol2 -fi mol2 -o charges2.mol2 -fo mol2 -c wc "
                    "-cf charges.txt -pf yes")
```
raised the error.
If you look at the above error raised, it can be understood that the "sanity-recheck" on the yielded "charged.mol2" file, is not passed. I believe that the error can be attributed to that the yielded mol2 file does not contain the net charge info.
In fact, when I add the option "-dr no", which disables the acdoctor mode, during the writing-out of the calculated charges, no errors will be raised.

So, here comes another issue: Now I have the AM1-BCC charges, how do I use them?
I have noticed that there is an option "charge_from_molecule" in the `forcefield.create_openmm_system()`.
However, it is noted that, here, RDKit is only considered, therefore support for "mol2" is not enabled/recommended.
How do I put the calculated charges into the "sdf" files, and let the Molecule.from_file() automatically recognize the partial charges?
Or, should I manually modify the molecule.partial_charges attribute? If so, how to do it and is it reliable (or error-prone)?

Thanks!

**Comment by j-wags** [comment]
Hi @Kael-the-Mad,

Thanks for the extensive write up. I have good news for you -- The OpenFF Toolkit 0.7.0 release (which came out 40 hours ago) should fix this issue (at least it makes your code run on my machine). It looks like you ran into the same issue as #492. Please let me know if upgrading to the newest toolkit version makes it work for you.

You can install it using:
```
conda create -n off070 -c conda-forge -c omnia openforcefield
conda activate off070
```

It's been a long time since our last release and many dependencies (including ambertools) have had major releases. That's why I suggest making a totally new conda environment instead of doing a normal upgrade.

> So, here comes another issue: Now I have the AM1-BCC charges, how do I use them?
> I have noticed that there is an option "charge_from_molecule" in the forcefield.create_openmm_system().

You would use it like this:
```
molecule = Molecule.from_file('mymol.sdf') 
forcefield = ForceField('openff-1.2.0.offxml') 
topology = Topology.from_molecules(molecule) 
omm_system = forcefield.create_openmm_system(topology, charge_from_molecules=[molecule]) 
```
So, you write something like `ff.create_openmm_system(topology, charge_from_molecules=[mol1, mol2])`, where you have already set the `molecule.partial_charges` attribute set on `mol1` and `mol2`. Then, any instances of `mol1` or `mol2` in the topology will have the charges from those Molecules assigned (skipping AM1-BCC)

> However, it is noted that, here, RDKit is only considered, therefore support for "mol2" is not enabled/recommended.
> How do I put the calculated charges into the "sdf" files, and let the Molecule.from_file() automatically recognize the partial charges?

The 0.7.0 OpenFF Toolkit also adds support for writing charges to SDF. They're stored in the `<atom.drop.PartialCharges>` SD data fields. So, now, when you read and write SDF files, the partial charges should stay on your molecules

Please let me know I covered all your questions, and if you can get the new OpenFF Toolkit running on your system :-)

**Comment by Kael-the-Mad** [comment]
Hi @j-wags ,
Thanks very much for the quick reply!
I am downloading the new released "openforcefield-0.7.0-py37-Linux-x86_64.sh", for installing it offline into my school's server.

The good news from you is really encouraging! 
I will firstly study the release notes of 0.7.0 and try it out. As soon as the test is finished, I will report it here.

**Comment by Kael-the-Mad** [comment]
Hi @j-wags ,
The adapted script can run without errors.
I have curiously checked the new openforcefield/utils/toolkits.py. I think the '"dr" "-n"' option is functioning here.

I have also noticed that there are several new partial charge methods available in the newly released 0.7.0 off.
Given that the sqm can be (extremely) slow on complex molecular structures, these new charge calculation methods may be handy. 
However, may I ask that is there a suggested priority or recommended "rank" or advantage over these methods? 
That is, whether the AM1-BCC is the best (though slow), and the others (e.g., the fast "gasteiger" and the "chargeIncrementMethod") are somehow not as recommended (given that simulations with existing AMBER/CHARMM-force fields could be the ultimate goal of using the open force field)? Or, where can I cite/read relevant scientific literature to support a certain choice of partial-charge methods?

Thanks!

**Comment by davidlmobley** [comment]
@Kael-the-Mad -- our force fields are designed to work with AM1-BCC. Support for other methods is to enable spin-off science/side projects, but if you're using one of our force fields we strongly recommend against changing the charge model.

`chargeIncrementMethod` is to enable us to do future BCC-like re-parameterizations as AM1-BCC is not the end of the story in that regard. 

Sorry to be brief/not fully answer your questions, but I have a rather busy day on this. I guess the big picture is -- you'll see a lot of things enabled by our infrastructure that are there to help us (or others) do future science, and thus are not yet the topic of publications and need future research.

**Comment by Kael-the-Mad** [comment]
@davidlmobley  
Thanks very much for answering my question. I will follow your recommendation.

@j-wags  
I sort of figure out that, the bottleneck step of the adapted script is the sqm calculation, which can be done separately with only the SDF files.
Thus, I simply used the following snippet (in a try-excerpt way)
```
xBinder = Molecule.from_file('*.sdf')
xBinder.assign_partial_charges('am1bcc')
xBinder.to_file('q*.sdf', file_format='sdf') 
```
Then, the generated "AM1-BCC"-charged SDF files will be directly used in the subsequent simulations.
I think my questions have been solved.
Thanks again.
